smack-core/src/main/java/org/jivesoftware/smack/util/NumberUtil.java

```diff
@@ -42,35 +42,6 @@ public static long requireUInt32(long value) {
         if (value > ((1L << 32) - 1)) {
             throw new IllegalArgumentException("unsigned 32-bit integers can't be greater than 2^32 - 1");
         }
-    }
```

Comment:
@Flowdalic I think I have messed it up here. I think it is possible to undo changes on `NumberUtils.java` using `git rebase` by using `drop`. But I don't want to mess it up further, so thought better to ask you before I make it any worse.

Reply:
The commits of this PR surely need cleanup, but the good news is that the diff looks good. So I'd simply fix the remaining issues and then squash everything into a single commit.